addons/mrp/static/src/widgets/mrp_production_components_x2many.js

```diff
@@ -3,8 +3,17 @@
 import { registry } from "@web/core/registry";
 import { X2ManyField } from "@web/views/fields/x2many/x2many_field";
 import { ListRenderer } from "@web/views/list/list_renderer";
+import { onPatched } from "@odoo/owl";
 
 export class MrpProductionComponentsListRenderer extends ListRenderer {
+    setup() {
+        super.setup(...arguments);
+
+        onPatched(() => {
```

Comment:
I don't agree with the current fix since this just resets the column width at each change and bypasses the job done by the framework just to be sure that the value will be recomputed after the column has become visible. I think it is important to pint point the precise reason why the columnWidth is not recomputed to find an appropriate fix.

Here is what I found during my investigation (reason of the issue):

When you edit the record by adding a new line this `useEffect` is triggered and the `keepColumnWidths` is set to true:
https://github.com/odoo/odoo/blob/4d4bedc1973bf7d6f2f9386d8cffab14ad890dd0/addons/web/static/src/views/list/list_renderer.js#L169-L176
When you confirm the MO a new column becomes visible and a call of the `freezeColumnWidths` is launched to reset and recompute the `columnWidiths` if necessary. However, in our case and because of the previous step this reset is skipped:
https://github.com/odoo/odoo/blob/4d4bedc1973bf7d6f2f9386d8cffab14ad890dd0/addons/web/static/src/views/list/list_renderer.js#L233-L236

The problem does not occur in 17.0 thanks to: https://github.com/odoo/odoo/pull/140663
This exact change should also work in 16.0 so that the most appropirate fix is probably just to backport this change (even thought this code is also dirty and bypasses part of the framework work).

You could perhaps mention these facts in the commit message for future references.

Reply:
Thank you for these precise explanations !
This fix has been backported in the Odoo 16 equivalent JS file.